packages/camera/camera/ios/Classes/QueueHelper.m

```diff
@@ -4,12 +4,26 @@
 
 #import "QueueHelper.h"
 
+const char *FLTCaptureSessionQueueSpecific = "capture_session_queue";
+
 @implementation QueueHelper
+
 + (void)ensureToRunOnMainQueue:(void (^)(void))block {
   if (!NSThread.isMainThread) {
     dispatch_async(dispatch_get_main_queue(), block);
   } else {
     block();
   }
 }
+
++ (dispatch_queue_t)createQueueWithLabel:(const char *)label specific:(const char *)specific {
```

Comment:
I generally don't like utility helpers that are only a few lines.  It essentially makes developers who are probably already familiar with gcd APIs now have to do the mental jump of understanding what the camera `createQueueWithLabel` method is doing.  This wraps a well-known standard library APIs around a bespoke helper method that doesn't do much and is more the letter than the spirit of DRY.

Reply:
I don't have a strong opinion here, so will update. 

On the one hand, I think the benefit of `createQueueWithLabel:specific` util function is that it combines 2 calls, so that the caller site won't forget to set the queue specific data; And the benefit of `isCurrentlyOnQueueWithSpecific:` util function is that it makes the assertion readable like english sentence (e.g. `NSAssert([QueueHelper isCurrentlyOnQueueWithSpecific:fooSpec]) vs NSAssert(dispatch_get_specific(fooSpec))`

But on the other hand you also made a very good point that these are 2 extra functions that developers have to learn.